code/game/data_huds.dm

```diff
@@ -245,22 +245,30 @@
 
 	hud_list[XENO_DEBUFF_HUD] = xeno_debuff
 
+	status_hud.overlays.Cut()
 	if(species.species_flags & IS_SYNTHETIC)
 		simple_status_hud.icon_state = ""
 		if(stat != DEAD)
 			status_hud.icon_state = "synth"
+			switch(round(health * 100 / maxHealth)) // special health HUD icons for damaged synthetics
+				if(-29 to 4) // close to overheating: should appear when health is less than 5
+					status_hud.icon_state = "synthsoftcrit"
+				if(-INFINITY to -30) // dying
+					status_hud.icon_state = "synthhardcrit"
 		else if(HAS_TRAIT(src, TRAIT_UNDEFIBBABLE))
 			status_hud.icon_state = "synthdnr"
 			return TRUE
 		else
 			if(!mind)
-				var/mob/dead/observer/G = get_ghost(TRUE)
-				if(!G)
+				var/mob/dead/observer/ghost = get_ghost(TRUE)
+				if(!ghost)
 					status_hud.icon_state = "synthdnr"
-				else
-					status_hud.icon_state = "synthdead"
-			else
-				status_hud.icon_state = "synthdead"
+					return TRUE
+				if(!ghost.client) // AFK ghost detected
+					status_hud.overlays += "dead_noclient"
```

Comment:
this is not afk, it means the player disconnected as a ghost, those are very different